conference/src/main/java/com/kyriba/conference/management/api/ScheduleController.java

```diff
@@ -30,78 +37,74 @@
+    List<PresentationResponse> presentations = stream(scheduleService.getSchedule().spliterator(), false)
+        .map(this::toPresentationResponse)
+        .collect(Collectors.toList());
+    return ScheduleResponse.builder().presentations(presentations).build();
   }
 
 
   @ApiOperation(value = "Add presentation in conference schedule")
   @PostMapping(value = "/presentations", produces = APPLICATION_JSON_UTF8_VALUE, consumes = APPLICATION_JSON_UTF8_VALUE)
   Long addPresentation(
-      @ApiParam(value = "Presentation create object", required = true) @RequestBody PresentationRequest presentation)
+      @ApiParam(value = "Presentation create object", required = true) @RequestBody PresentationRequest presentationRequest) throws
+      LinkedEntityNotFound
   {
-    return 55L;
+    return scheduleService.addPresentation(presentationRequest);
   }
 
 
   @ApiOperation(value = "View presentation information")
   @GetMapping(value = "/presentations/{id}", produces = APPLICATION_JSON_UTF8_VALUE)
-  PresentationResponse getPresentation(
-      @ApiParam(value = "Presentation identity", required = true) @PathVariable String id)
+  ResponseEntity<PresentationResponse> getPresentation(
+      @ApiParam(value = "Presentation identity", required = true) @PathVariable Long id)
   {
-    return PresentationResponse.builder()
-        .hall(123L)
-        .startTime(LocalTime.of(11, 30))
-        .endTime(LocalTime.of(12, 30))
-        .topic(new TopicDto("Spring Data REST", "Andy Wilkinson"))
-        .build();
+    return ResponseEntity.of(scheduleService.getPresentation(id).map(this::toPresentationResponse));
   }
 
 
   @ApiOperation(value = "Remove presentation from conference schedule")
   @DeleteMapping(value = "/presentations/{id}", consumes = APPLICATION_JSON_UTF8_VALUE)
+  @ResponseStatus(value = NO_CONTENT)
   void removePresentation(
-      @ApiParam(value = "Presentation identity", required = true) @PathVariable String id)
+      @ApiParam(value = "Presentation identity", required = true) @PathVariable Long id)
   {
-
+    scheduleService.deletePresentation(id);
   }
 
 
   @ApiOperation(value = "Change presentations parameters")
   @PutMapping(value = "/presentations/{id}", consumes = APPLICATION_JSON_UTF8_VALUE)
   void updatePresentation(
-      @ApiParam(value = "Presentation identity", required = true) @PathVariable String id,
-      @ApiParam(value = "Presentation change object", required = true) @RequestBody PresentationRequest presentation)
+      @ApiParam(value = "Presentation identity", required = true) @PathVariable Long id,
+      @ApiParam(value = "Presentation change object", required = true) @RequestBody PresentationRequest presentationRequest) throws
+      EntityNotFound, LinkedEntityNotFound
   {
-
+    scheduleService.updatePresentation(id, presentationRequest);
   }
 
 
-  private List<PresentationResponse> buildPresentations()
+  private PresentationResponse toPresentationResponse(Presentation presentation)
```

Comment:
This is service logic

Reply:
see https://github.com/gariklolkin/conference/pull/13/files#r298810884